### PR TITLE
fix(lib): detect decorator in abstract component (ngneat#97)

### PR DESCRIPTION
**Pull request by martinbianchi**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/until-destroy/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Inherit from abstract component throws an error

Issue Number: #97 

## What is the new behavior?
You should be able to inherit from an abstract component and everything should work

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

I'm Working on the tests. I submit the PR to show the approach on this issue, I've tested with the Reproducible example on the issue and it seems to work. I had some troubles trying to run the integration test (Them didn't work in my environment wirh no changes in the repo). But as I said before the idea is to check the approach I take and get your feedback.

Thanks.

### CODE REVIEW

File: src/lib/ivy.ts [hunk 1]
```diff
@@ -26,13 +26,15 @@ export interface PipeType<T> extends Type<T> {
 
 // As directive and component definitions are considered private API,
 // so those properties are prefixed with Angular's marker for "private".
+// An abstract component is not decorated by angular, so we should return the prototype.
```
Comment (arturovt) [inline src/lib/ivy.ts:29]: Abstract classes are decorated by decorators especially if they have declared bindings. Angular 9 introduced an empty Directive decorator that can be applied specifically for abstract classes.

Comment (martinbianchi) [reply src/lib/ivy.ts:29]: Okey, I will check that. The reproducible example has an Abstract Component with no decorators. We should think if the idea is make the decorator mandatory for abstract component or allow either

Comment (martinbianchi) [reply src/lib/ivy.ts]: Hey @arturovt I checked about the empty Directive decorator. Is mandatory when we want to use some Angular features in an Abstract Component. I updated the PR to support both, decorated and no decorated abstract components.

**Comment by martinbianchi** [comment]
Hi, I'm a bit lost here with the tests. If anyone could guide me a bit would be great!

**Comment by NetanelBasal** [comment]
@martinbianchi what's the issue with the specs?

**Comment by NetanelBasal** [comment]
#119 fix this issue.

**Comment by arturovt** [comment]
Closing this since `8.0.0` has been published.

**Pull request closed without merging**